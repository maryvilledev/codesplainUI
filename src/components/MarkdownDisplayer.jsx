import React, { PropTypes } from 'react';
import MarkdownRenderer from 'react-markdown-renderer';
import { Card } from 'material-ui';

import markdownRendererOptions from '../util/markdown-renderer-options';

const styles = {
  card: {
    padding: '0.5rem',
  },
};

const MarkdownDisplayer = ({ annotation }) => (
  <Card style={styles.card} >
    <MarkdownRenderer
      markdown={annotation}
      options={markdownRendererOptions}
    />
  </Card>
);

MarkdownDisplayer.propTypes = {
  annotation: PropTypes.string.isRequired,
};

export default MarkdownDisplayer;
