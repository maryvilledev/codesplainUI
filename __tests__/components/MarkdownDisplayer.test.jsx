import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import MarkdownDisplayer from '../../src/components/MarkdownDisplayer';

const defaultProps = {
  annotation: '',
};

describe('<MarkdownDisplayer />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  describe('snapshots', () => {
    describe('matches snapshot', () => {
      it('matches snapshot', () => {
        const wrapper =
          shallowWithContext(<MarkdownDisplayer {...defaultProps} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('prop: annotation', () => {
      it('is forwarded to MarkdownRenderer', () => {
        const annotation = 'No jumping in the sewer!';
        const wrapper = shallowWithContext(
          <MarkdownDisplayer annotation={annotation} />,
        );
        const markdownRenderer = wrapper.find('MarkdownRenderer');
        expect(markdownRenderer.prop('markdown')).toEqual(annotation);
      });
    });
  });
});
