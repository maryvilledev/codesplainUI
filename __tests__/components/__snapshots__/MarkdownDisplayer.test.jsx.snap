exports[`<MarkdownDisplayer /> snapshots matches snapshot matches snapshot 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
  style={
    Object {
      "padding": "0.5rem",
    }
  }>
  <MarkdownRenderer
    markdown=""
    options={
      Object {
        "highlight": [Function],
        "langPrefix": "hljs language-",
      }
    } />
</Card>
`;
